modules/system/classes/UpdateManager.php

```diff
@@ -365,21 +365,29 @@ public function uninstall()
     }
 
     /**
-     * Asks the gateway for the lastest build number and stores it.
+     * Asks the composer/gateway for the lastest build number and stores it.
      * @return void
      */
     public function setBuildNumberManually()
     {
-        $postData = [];
+        // Use installed.json to detect latest version
+        $composerInstalled = base_path('vendor/composer/installed.json');
+        if (file_exists($composerInstalled)) {
+            $packages = json_decode(file_get_contents($composerInstalled), true);
```

Comment:
@Samuell1 I'm concerned we will run into memory issues reading this file entirely, as Composer dependencies could be huge depending on the number of plugins used and various other factors. Given that this command is run in a command-line environment, would it be safe enough to just run `composer show october/system` through `shell_exec` and get the version that way?

Reply:
@bennothommo would depend on having composer installed, not all october environments or projects use composer, including the october installer itself does not use composer.

Reply:
@prhost the normal "set build" functionality is underneath this code block, and still works for non-Composer based installations.

Reply:
I meant that running the `composer show october/system` command via shell will depend on whether OS has composer bin installed.

Reply:
@prhost True, but I think it's reasonable enough to assume that the binary is installed if the `vendor/composer` folder exists. We could fall back to reading the installed.json file if it doesn't.

Reply:
installed trough ocms installer and vendor with installed.json is there

Reply:
I would particularly vote to create a metadata or constant with the october version in package `october/system` and commit and keep it up to date when releasing a version.

What do you think? Because there are tools I for example step into my project to eliminate unwanted waste and files, making the project lighter when I create an artifact or release and go into production. Reading this into a json file is creating an unnecessary dependency.

Reply:
I dont know but installed,json is internal composer file that is needed for composer to work idk if any tool can remove it.

Reply:
@Samuell1 thanks for checking the installer. Would you mind testing the memory usage of just decoding the JSON file as you had it before, with a whole heap of dependencies installed? If the memory doesn't become too high, I'm fine with leaving it as is.

@prhost If, for example, you delete the `installed.json` file, the system will simply use the old method of pinging the October server for the build number. There's no explicit dependency on the `installed.json` file - it's just a shortcut for determining the build number if it's available.

Reply:
@bennothommo yes

Its not only shortcut but exact version of your ocms installed locally. I seen many times issue that version installed locally wasnt same as set in administration.

Reply:
This is exactly why the build version should be bundled with some file in the repository. and @Samuell1 does work october without installed.json, what doesn't work is composer, but in production environment you don't need composer, so I'm talking about dependency.

A `define("OCTOBER_VERSION", 420);` like laravel does would already solve all that.

But this is all my opinion, @bennothommo decides whatever he wants (:

Reply:
You cant run ocms without vendor folder.

Reply:
While I'm not opposed to a constant defining the version, it does mean one extra manual step for us when pushing out a new version of October. I would prefer if it was automated in some way.

Reply:
Just tested that json_decode it spikes to 18MB of ram :D

Reply:
> You cant run ocms without vendor folder.

Not what I meant hehe. october does not depend on composer, it depends on its vendor dependencies, the installed.json file is not a file that october or vendor depends on, deletes and your application will work normally.

But I agree with @bennothommo , updating constant is yet another process when the release is released. But the form that is currently not the correct one, as the query to webservice will always return the last version released, which may not be the current correspondent.

Perhaps the opinions of others could help.

Reply:
It depends how it can be most simple from both sides implemented.

Reply:
Any ideas @bennothommo what to use to prevent big memory usage from parsing that json?

Reply:
Use non-memory based parsers 😄 `json_decode` -> get everything inside of memory and proceed.
Like: https://github.com/halaxa/json-machine or https://github.com/salsify/jsonstreamingparser

Reply:
@w20k does exists something in ocms like that without needing to add another package?

Reply:
The only way you could do it is a raw string parsing (regex or by char). 
I don't think there is an option out of the box in PHP (might be wrong, brain exploded from a different language I'd to use in past few weeks).

Reply:
@Samuell1 18MB should be fine if that's all that's happening in the request, most environments won't have anything less than 128MB allotted.

Reply:
@LukeTowers that size depends on size of that json, but that 18 is just once and it should be fine, but maybe ben knows better solution.

Reply:
@Samuell1 Yeah I'm happy with 18MB. Even if it was 4 times the size, we're looking at maybe 72MB of memory usage, which is not too bad. If it becomes common, we'll look at some form of streaming or another method of getting the version.